File: store_credit/store_credit.scala
object StoreCredit extends Application {	
	type Item = (Int, Int)
	
	def solve(credit: Int, items: List[Item]): (Item, Item) = {
		val it = items.iterator
		val head = it.next
		val m = it.find((e: Item) => head._1 + e._1 == credit)
		if (m != None) {
			return (head, m.get)
		}
		solve(credit, items.slice(1, items.length))
	}
	
	val N = readInt()
	for (n <- List.range(1, N + 1)) {
		val credit = readInt()
		val count = readInt()
		val values = readLine.split(" ").map((s: String) => s.toInt)
		val items = values.toList.zip(List.range(1, count + 1))
		val r = solve(credit, items)
		val indice = List(r._1._2, r._2._2).sortWith((a, b) => a < b).mkString(" ")
		println("Case #" + n + ": " + indice)		
	}
}

File: store_credit/StoreCredit.java
import java.io.*;
import java.util.*;

class Item {
	int value, index;
	Item(int v, int i) { value = v; index = i; }
}

public class StoreCredit {
	static Item[] solve(int credit, List<Item> items) {
		Item head = items.remove(0);
		for (Item e: items) {
			if (e.value + head.value == credit) {
				return new Item[]{head, e};
			} 
		}
		return solve(credit, items);
	}
	
	public static void main(String[] args) throws Exception {
		BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
		int N = Integer.parseInt(in.readLine());
		for (int n = 1; n <= N; n++) {
			int credit = Integer.parseInt(in.readLine());
			in.readLine();
			ArrayList<Item> items = new ArrayList<Item>();
			int index = 0;
			for (String s: in.readLine().split(" ")) {
				index++;
				int value = Integer.parseInt(s);
				if (value < credit) {
					items.add(new Item(value, index));					
				}
			};
			Item[] r = solve(credit, items);
			int[] indice = {r[0].index, r[1].index};
			Arrays.sort(indice);
			System.out.println("Case #" + n + ": " + indice[0] + " " + indice[1]);
		}
	}
}